### PR TITLE
Add text embedder

### LINKED ISSUE #760: Add TextEmbedder Task
## 🚀 Feature
Using https://github.com/UKPLab/sentence-transformers create a Flash Task for embedding sentences. Sentence Transformers also supports fine-tuning the model onto your own set of data, to improve semantic embedding representations.

### Motivation

It's common to require text search, and sentence-transformers provides great results + an API to fine-tune/predict to get sentence representations out. It could also be paired with [FAISS](https://github.com/facebookresearch/faiss) really simply to provide fast/accurate search results across large datasets.

### PR DESCRIPTION
**Pull request by abhijithneilabraham**
Fixes #760

### CODE REVIEW

File: flash/text/embedding/model.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    def training_step(self, batch: Any, batch_idx: int) -> Any:
+        raise NotImplementedError("Training a `SentenceEmbedder` is not supported. Use a different text task instead.")
+
+    def validation_step(self, batch: Any, batch_idx: int) -> Any:
+        raise NotImplementedError(
+            "Validating a `SentenceEmbedder` is not supported. Use a different text task instead."
+        )
+
+    def test_step(self, batch: Any, batch_idx: int) -> Any:
```
Comment (tchaton) [inline flash/text/embedding/model.py:84]: What about the predict_step ?

File: flash_examples/text_embedder.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+datamodule = TextClassificationData.from_lists(
+    predict_data=[
+        "Turgid dialogue, feeble characterization - Harvey Keitel a judge?.",
+        "The worst movie in the history of cinema.",
+        "I come from Bulgaria where it 's almost impossible to have a tornado.",
+    ]
+)
+
+# 2. Load a previously trained SentenceEmbedder
+model = SentenceEmbedder(backbone="sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2")
```
Comment (tchaton) [inline flash_examples/text_embedder.py:32]: Would you mind adding some tests to increase the coverage ?

**Review by tchaton — COMMENTED** [review]
Great work ! Almost there !

File: flash/text/embedding/model.py (hunk 1 above)
Comment (abhijithneilabraham) [reply flash/text/embedding/model.py:84]: We used the forward to generate the embeddings. As for the predict step, it relies on the predict_step implementation in the base task.

File: flash_examples/text_embedder.py (hunk 2 above)
Comment (abhijithneilabraham) [reply flash_examples/text_embedder.py:32]: Yes. On it now.

Comment (abhijithneilabraham) [reply flash_examples/text_embedder.py:32]: @tchaton  I just commited a test but I am not sure if it is the right way you want it to be. Please do a review and let me know how I can improve.

File: flash/text/embedding/model.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+if _TEXT_AVAILABLE:
+    from sentence_transformers.models import Pooling
+
+    Pooling = print_provider_info("Pooling", _SENTENCE_TRANSFORMERS, Pooling)
+
+logger = logging.getLogger(__name__)
+
+
+class SentenceEmbedder(Task):
```
Comment (ethanwharris) [inline flash/text/embedding/model.py:38]: Shall we rename this to `TextEmbedder`?

File: requirements.txt [hunk 4]
```diff
@@ -7,3 +7,4 @@ pyDeprecate
 pandas>=1.1.0
 jsonargparse[signatures]>=3.17.0
 click>=7.1.2
+tqdm
```
Comment (ethanwharris) [inline requirements.txt:10]: Do we need this? I think `tqdm` is included in PL requirements

File: tests/text/embedding/test_model.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    predict_data=[
+        "Turgid dialogue, feeble characterization - Harvey Keitel a judge?.",
+        "The worst movie in the history of cinema.",
+        "I come from Bulgaria where it 's almost impossible to have a tornado.",
+    ]
+)
+
+# ==============================
+
+TEST_BACKBONE = "sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2"  # super small model for testing
```
Comment (ethanwharris) [inline tests/text/embedding/test_model.py:35]: This looks like quite a big model, perhaps `sentence-transformers/all-MiniLM-L6-v2` would be better (seems around a quarter of the size)

**Review by ethanwharris — COMMENTED** [review]
Awesome, looking good :smiley: A few comments suggestions. Also, could you update `CHANGELOG.md`?

File: flash_examples/text_embedder.py [hunk 6]
```diff
@@ -27,9 +27,10 @@
 )
 
 # 2. Load a previously trained SentenceEmbedder
-model = SentenceEmbedder(backbone="sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2")
+model = SentenceEmbedder(backbone="sentence-transformers/all-MiniLM-L6-v2")
```
Comment (abhijithneilabraham) [inline flash_examples/text_embedder.py:30]: Done! @ethanwharris

File: requirements.txt (hunk 4 above)
Comment (abhijithneilabraham) [reply requirements.txt:10]: yeah I suppose we don't

File: flash_examples/text_embedder.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+
+import flash
+from flash.core.data.utils import download_data
```
Comment (ethanwharris) [inline flash_examples/text_embedder.py:17]: This line can be removed :smiley:
```suggestion
```

Comment (abhijithneilabraham) [reply flash_examples/text_embedder.py]: Done! Also changed the name from `SentenceEmbedder` to `TextEmbedder`

**Review by ethanwharris — APPROVED** [review]
Awesome, LGTM :smiley:

**Comment by abhijithneilabraham** [comment]
@tchaton

**Review by tchaton — APPROVED** [review]
LGTM ! Awesome work !

**Pull request merged**